scripts/gt_concordance.py

```diff
@@ -0,0 +1,551 @@
+    lost = sorted(lost, key=lambda x:(int(x[0]) if x[0] != 'X' else x[0], int(x[1])))
+
+    # recording results
+    result = '{}\n'.format(sample)
+    result += 'Chrom Pos A1_seq A2_seq A1_maf A2_maf\n'
+    result += 'Mismatches: {}\n'.format(len(mismatches))
+    result += '\n'.join([' '.join(mismatch) for mismatch in mismatches])
+    result += '\nLost: {}:\n'.format(len(lost))
+    result += '\n'.join([' '.join(lost_snp) for lost_snp in lost])
+    result += '\n'
+    result += '\nLost snps: {}\n'.format(len(lost))
+    result += 'Total number of matches: {} / {} / {}\n'.format(len(matches), len(vcf_data)-len(lost), len(vcf_data))
+    result += 'Percent matches {}%\n'.format(percent_matches)
+
+
+    # path should exist (if we came to this point), but checking anyway
+    output_path = os.path.join(config.get('ANALYSIS_PATH'), project, 'piper_ngi/03_genotype_concordance')
+    if os.path.exists(output_path):
+        # create .conc file
+        with open(os.path.join(output_path, '{}.conc'.format(sample)), 'w+') as conc_file:
+            conc_file.write(result)
+
+    if len(matches) + len(mismatches) + len(lost) != len(vcf_data):
+        log.warning('CHECK RESULTS!! Numbers are incoherent. Total number of positions: {}, matches: {}, mismatches: {}, lost: {}'.format(len(vcf_data), len(matches), len(mismatches), len(lost)))
+
+    if len(lost) >= 30:
+        log.warning('Too few positions in VCF file!! Failed to caclulate concordance')
+    return percent_matches
+
+def run_gatk(sample, config):
+    project = sample.split('_')[0]
+    ANALYSIS_PATH = config.get('ANALYSIS_PATH')
+    # the path has been already checked, but checking again
+    if os.path.exists(ANALYSIS_PATH):
+        bamfile = os.path.join(ANALYSIS_PATH, project, 'piper_ngi/05_processed_alignments/{}.clean.dedup.bam'.format(sample))
+        if not os.path.exists(bamfile):
+            log.error('bamfile does not exist! {}'.format(bamfile))
+            return None
+        project = sample.split('_')[0]
+        # the path has been already checked
+        output_file = os.path.join(ANALYSIS_PATH, project, 'piper_ngi/03_genotype_concordance', "{sample}.vcf".format(sample=sample))
+        options = """-T UnifiedGenotyper  -I {bamfile} -R {gatk_ref_file} -o {sample}  -D {gatk_var_file} -L {interval_file} -out_mode EMIT_ALL_SITES """.format(
+                bamfile=bamfile,
+                sample=output_file,
+                interval_file=config.get('INTERVAL_FILE'),
+                gatk_ref_file=config.get('GATK_REF_FILE'),
+                gatk_var_file=config.get('GATK_VAR_FILE'))
+        full_command = 'java -Xmx6g -jar {} {}'.format(config.get('GATK_PATH'), options)
+        try:
+            subprocess.call(full_command.split())
+        except:
+            pass
+        else:
+            return output_file
+
+def update_charon(sample_id, status, concordance=None):
```

Comment:
suggestion: call this function `update__genotype_status_charon ` or something similar,`update_charon ` is a bit too general and confusing